### PR TITLE
39 toggle for list view on home home list

### PR DESCRIPTION
**Pull request by JustinDotMorris**

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+  buttonImgAlt,
+  buttonImgClass,
+}) => {
   let buttonClass = `button button--${buttonName}`;
+  let buttonImageName = `button--image--${buttonImgClass}`;
 
   return (
     <button className={buttonClass} onClick={onClick}>
       {buttonText}
+      <img className={buttonImageName} src={buttonImgUrl} alt={buttonImgAlt} />
```
Comment (DanForder) [inline src/components/Button/Button.jsx:17]: Not every button needs an ``, so perhaps hide this if there is no `buttonImgUrl`?

File: src/components/Button/Button.scss [hunk 2]
```diff
@@ -12,10 +12,24 @@
...
 
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
   &:active {
     border: 0.2rem solid $white;
   }
 }
+.button--image--discover-icon{
```
Comment (DanForder) [inline src/components/Button/Button.scss:23]: You have two classes here, but I don't think it's modifying an existing class?

`.button__discover-icon` ?

File: src/pages/Home/Home.test.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import { customRender } from "../../utils/testUtils";
+import { screen } from "@testing-library/react";
+
+it("should render a button on the screen", () => {
+  customRender(<Button buttonImgClass={"discover-icon"} />);
+
+  const buttononscreen = screen.getByRole("button");
+
+  expect(buttononscreen).toBeInTheDocument();
+});
```
Comment (DanForder) [inline src/pages/Home/Home.test.jsx:11]: Why are we testing the button in the Home test? Probably should be testing the `<Home>` component!

**Review by DanForder — CHANGES_REQUESTED** [review]
Lots of `--undefined` classnames coming through, due to the button's image always showing

File: src/pages/Home/Home.test.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import Home from "../../pages/Home/Home";
+import { customRender } from "../../utils/testUtils";
+import { screen } from "@testing-library/react";
+
+it("should render a button on the screen", () => {
+  customRender(<Home />);
+
+  const buttononscreen = screen.getByText("List View");
```
Comment (DanForder) [inline src/pages/Home/Home.test.jsx:8]: if you're looking for a button, let's use `getByRole` and use the button role

File: src/components/Button/Button.scss [hunk 5]
```diff
@@ -11,11 +11,23 @@
...
 }
 
+.button--map-navigation{
+  background-color: $white;
+  color: $black;
+  width: 3rem;
+  height: 3rem;
+  font-size: 1rem;
+}
+.button-plus,.button-minus{
```
Comment (DanForder) [inline src/components/Button/Button.scss:27]: should these be `.button--plus`, `.button--minus` ?

**Review by DanForder — CHANGES_REQUESTED** [review]

File: src/pages/Home/Home.jsx [hunk 6]
```diff
@@ -11,6 +12,9 @@ const Home = () => {
   }
   return (
     <Layout isWithMenu={true}>
+      <Button buttonName="map-navigation button--plus" buttonText="+" />
+      <Button buttonName="map-navigation button--minus" buttonText="-" />
```
Comment (DanForder) [inline src/pages/Home/Home.jsx:16]: Are these plus/minus styles being applied? I think it needs to passed through as `button-minus` and `button-plus`. the Button component adds the `--` itself

**Review by DanForder — CHANGES_REQUESTED** [review]

**Review by DanForder — APPROVED** [review]

**Review by Cris2508 — CHANGES_REQUESTED** [review]
Hi, 2 changes needs to be done:
1. the border size is 0.125rem;
2. the buttons should be vertical and in the same line.

**Review by Cris2508 — APPROVED** [review]
Hi, 2 changes needs to be done:
1. the border size is 0.125rem;
2. the buttons should be vertical and in the same line.

**Pull request merged**